kyuubi-common/src/main/scala/org/apache/kyuubi/util/FileExpirationUtils.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.util
+
+import java.nio.file.{Path, Paths}
+import java.util
+import java.util.Collections
+import java.util.concurrent.atomic.AtomicBoolean
+
+import scala.collection.convert.ImplicitConversions.`collection asJava`
+
+import org.apache.kyuubi.Utils
+
+object FileExpirationUtils {
+  private lazy val isCleanupShutdownHookInstalled = {
+    addFilesCleanupOnExitShutdownHook()
+    new AtomicBoolean(true)
+  }
+
+  private lazy val files: util.Set[String] =
+    Collections.synchronizedSet(new util.LinkedHashSet[String]())
```

Comment:
it's weird to maintain state in a Utils class, can we merge it into the Service class? BTW, we don't need to use an external collection to track those files, a Cache instance is sufficient, see https://github.com/apache/kyuubi/pull/6319/files

Reply:
It's a list not a state? Inside the Java util's `Files.deleteOnExit` method, there's a file list maintained in `DeleteOnExitHook`. So it is a similar design here.

Reply:
In fact, the `java.io.*` classes were one of the most controversial designs in the Java standard library, and the `java.nio.*` classes were later introduced in subsequent versions. Sometimes the JDK standard library does not follow common design patterns, especially parts closely related to the OS platform.

I would suggest moving TempFileManager to the common module, and centralizing the logic into it. Guava dep is not an issue for me, as it has no real effects on the leaf modules, e.g. `kyuubi-server`, `spark-sql-engine`

Reply:
BTW, I'm investigating the Spark Connect support, which based on the gRPC protobuf, Guava will eventually land common, and everywhere ...